open-craft/openedx-completion-aggregator
# Update aggregators whenever a block completion is updated.

**Description:** This adds the logic to update aggregators whenever a new block completion is created.

Manual test instructions are a little complex.  I'll update them in the next hour or so.

**Merge deadline:** None

**Installation instructions:** 

1.  Checkout edx/edx-platform to the cliff/aggregator-settings branch
2.  Run paver install_python_prereqs

**Testing instructions:**

1.  Set the completion.enable_completion_tracking waffle switch.
2.  Delete existing BlockCompletion objects:

        >>> from completion.models import BlockCompletion
        >>> BlockCompletion.objects.all().delete()

3.  Visit some pages in a course.
4.  Check that completion objects were created.

        >>> BlockCompletion.objects.all()

5. Check that Aggregators were created for course, chapter, sequential, and vertical blocks, and have sensible values.

        >>> from completion_aggregator.models import Aggregator
        >>> Aggregator.objects.all()

**Reviewers:**

FYI: @bradenmacdonald 

**Merge checklist:**

**Post merge:**

**Author concerns:**
